### DISCUSSION TITLE
Error: no authentication details supplied

### DISCUSSION THREAD

**Discussion by kpatelpanw** [jfrog/terraform-provider-artifactory#529] [Q&A] [answered]
Hi Team, I am trying to use below config and seeing "Error: no authentication details supplied" error. 
access_token is mandatory field ? Mainly the artifact (exist in jfrog artifactory repo) has anonymous read set, so anyone can access it without any authentication. How can I achieve the same for Artifactory provider ? I don;t want to specify any token. I see the same result with I set access_token  = "". Any idea ? 

```
# Required for Terraform 0.13 and up (https://www.terraform.io/upgrade-guides/0-13.html)
terraform {
  required_providers {
    artifactory = {
      source  = "registry.terraform.io/jfrog/artifactory"
      version = "2.6.17"
    }
  }
}

# Configure the Artifactory provider
provider "artifactory" {
  url           = "${var.artifactory_url}/artifactory"
  #access_token  = ""
}
```

=========
Running command: terraform plan

Error: no authentication details supplied

  on providers.tf line 56, in provider "artifactory":
  56: provider "artifactory" {
==========

**Comment by alexhung** [comment, marked as answer]
Sorry for the late reply. Currently the `access_token` attribute is required. This is because the majority of the resources/data sources requires the token to work.

**Comment by wyarde** [comment]
@kpatelpanw I ran into the same. You can get it to work without authentication by tricking the provider by setting `access_token` to `";"`.
```
provider "artifactory" {
  url = "${var.artifactory_url}/artifactory"
  check_license = false
  access_token = ";"
}
```
Hope it helps!
